src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -214,8 +218,25 @@ public CRAMFileReader(final InputStream stream,
     public CRAMFileReader(final File cramFile,
                           final File indexFile, final CRAMReferenceSource referenceSource,
                           final ValidationStringency validationStringency) throws IOException {
-        this(new FileInputStream(cramFile), indexFile, referenceSource, validationStringency);
+        this(new FileInputStream(cramFile),
+                wrapIndexFile(cramFile, indexFile),
```

Comment:
It would be an incremental step in minimizing the number of code paths (and exception propagation strategies) for creating SeekableStreams in this code. There are several bugs in the index/iterator state management in this code (some of which can be seen [here](https://github.com/samtools/htsjdk/issues?q=is%3Aissue+is%3Aopen+label%3Acram)). Ultimately, this needs a bigger refactoring anyway. At a minimum, if you keep this method, can you rename it to something more specific - maybe getStreamForIndex ? Thx.